storagenode/console/consoleserver/server.go

```diff
@@ -127,163 +105,108 @@ func (server *Server) Close() error {
+func (server *Server) satelliteHandler(wr http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	defer mon.Task()(&ctx)(nil)
 
-	err = server.service.CheckVersion(ctx)
-	if err != nil {
-		return response, err
+	if req.Method != http.MethodGet {
+		http.Error(wr, "method not allowed", http.StatusMethodNotAllowed)
+		return
 	}
 
-	bandwidthChartData, err := server.getBandwidthChartData(ctx, satelliteID)
+	satelliteID, err := storj.NodeIDFromString(strings.TrimLeft(req.URL.Path, "/api/satellite/"))
 	if err != nil {
-		return response, err
+		server.writeError(wr, http.StatusBadRequest, Error.Wrap(err))
+		return
 	}
 
-	// TODO: uncomment in future, when caching will be implemented
-	// diskSpaceChartData, err := server.getDiskSpaceChartData(ctx, satelliteID, satellites)
-	// if err != nil {
-	// 	return response, err
-	// }
-
-	uptime := server.service.GetUptime(ctx)
-	nodeID := server.service.GetNodeID(ctx)
-
-	// TODO: uncomment in future, when caching will be implemented
-	// if satelliteID != nil {
-	// 	satelliteStats, err := server.service.GetStatsFromSatellite(ctx, *satelliteID)
-	// 	if err != nil {
-	// 		return response, err
-	// 	}
-	//
-	// 	response.UptimeCheck = satelliteStats.UptimeCheck
-	// 	response.AuditCheck = satelliteStats.AuditCheck
-	// }
-
-	response.DiskSpace = *space
-	response.Bandwidth = *usage
-	response.WalletAddress = walletAddress
-	response.VersionInfo = versionInfo
-	response.IsLastVersion = true
-	response.Uptime = uptime
-	response.NodeID = nodeID.String()
-	response.Satellites = satellites
-	response.BandwidthChartData = bandwidthChartData
-	//response.DiskSpaceChartData = diskSpaceChartData
-
-	return response, nil
-}
+	if err = server.service.VerifySatelliteID(ctx, satelliteID); err != nil {
+		server.writeError(wr, http.StatusNotFound, Error.Wrap(err))
+		return
+	}
 
-func (server *Server) getBandwidth(ctx context.Context, satelliteID *storj.NodeID) (_ *console.BandwidthInfo, err error) {
-	if satelliteID != nil {
-		return server.service.GetBandwidthBySatellite(ctx, *satelliteID)
+	data, err := server.service.GetSatelliteData(ctx, satelliteID)
+	if err != nil {
+		server.writeError(wr, http.StatusInternalServerError, Error.Wrap(err))
+		return
 	}
 
-	return server.service.GetUsedBandwidthTotal(ctx)
+	server.writeData(wr, data)
 }
 
-func (server *Server) getBandwidthChartData(ctx context.Context, satelliteID *storj.NodeID) (_ []console.BandwidthUsed, err error) {
-	from, to := date.MonthBoundary(time.Now().UTC())
+// jsonOutput defines json structure of api response data.
+type jsonOutput struct {
+	Data  interface{} `json:"data"`
+	Error string      `json:"error"`
+}
 
-	if satelliteID != nil {
-		return server.service.GetDailyBandwidthUsed(ctx, *satelliteID, from, to)
-	}
+// writeData is helper method to write JSON to http.ResponseWriter and log encoding error.
+func (server *Server) writeData(wr http.ResponseWriter, data interface{}) {
```

Comment:
convention is w and r here as well

Reply:
this is not handler

Reply:
It's still managing the same method argument, so changing `w` to `wr` could be confusing and make me think they are different things.